File: Backend/TableTracker.Domain/DataTransferObjects/RestaurantDTO.cs
using System.Collections.Generic;

using TableTracker.Domain.Enums;

//TODO change Cuisines Type
namespace TableTracker.Domain.DataTransferObjects
{
    public class RestaurantDTO
    {
        public long Id { get; set; }

        public double CoordX { get; set; }

        public double CoordY { get; set; }

        public float Rating { get; set; }

        public string PriceRange { get; set; }

        public int NumberOfTables { get; set; }

        public RestaurantType Type { get; set; }

        public Discount Discount { get; set; }

        public ICollection<TableDTO> Tables { get; set; }

        public ICollection<WaiterDTO> Waiters { get; set; }

        public ICollection<string> Cuisines { get; set; }

        public FranchiseDTO Franchise { get; set; }

        public LayoutDTO Layout { get; set; }
    }
}

File: Frontend/src/app/models/dtos/restaurant.dto.ts
import { Discount } from "../enums/discount.enum"
import { RestaurantType } from "../enums/restaurant.enum"
import { CuisineDTO } from "./cuisine.dto"
import { ImageDTO } from "./image.dto"

export interface RestaurantDTO {
    id: number,
    address: string,
    name: string,
    description: string,
    rating: number,
    priceRange: number,
    numberOfTables: number
    type: RestaurantType,
    discount: Discount,
    mainImage: ImageDTO,
    cuisines: CuisineDTO[],
}
